pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.model.KitRequest;
+import org.broadinstitute.dsm.model.KitRequestSettings;
+import org.broadinstitute.dsm.model.KitType;
+import org.broadinstitute.dsm.util.DDPKitRequest;
+import org.broadinstitute.dsm.util.EasyPostUtil;
+import org.broadinstitute.lddp.util.DeliveryAddress;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+public class NotPepperKitCreationService {
+
+    public final static String ADDRESS_VALIDATION_ERROR = "UNABLE_TO_VERIFY_ADDRESS";
+    public final static String UNKNOWN_KIT_TYPE = "UNKNOWN_KIT_TYPE";
+    public final static String UNKNOWN_STUDY = "UNKNOWN_STUDY";
+    public final static String MISSING_JUNIPER_KIT_ID ="MISSING_JUNIPER_KIT_ID";
+    public final static String MISSING_JUNIPER_PARTICIPANT_ID ="MISSING_JUNIPER_PARTICIPANT_ID";
+    public final static String JUNIPER ="JUNIPER";
+    public final static String JUNIPER_UNDERSCORE ="JUNIPER_";
+
+    public KitResponse createNonPepperKit(JuniperKitRequest juniperKitRequest, String studyGuid, String kitTypeName) {
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID())) {
+            return new KitResponse(MISSING_JUNIPER_PARTICIPANT_ID, juniperKitRequest.getJuniperKitId());
+        }
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperKitId())) {
+            return new KitResponse(MISSING_JUNIPER_KIT_ID, null);
+        }
+        Optional<DDPInstanceDto> maybeDdpInstanceDto = new DDPInstanceDao().getDDPInstanceByStudyGuid(studyGuid);
+        if (maybeDdpInstanceDto.isEmpty()) {
+            return new KitResponse(UNKNOWN_STUDY, juniperKitRequest.getJuniperKitId());
+        }
+        DDPInstanceDto ddpInstanceDto = maybeDdpInstanceDto.get();
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(ddpInstanceDto.getInstanceName(), "juniper_study");
+
+        if (!ddpInstance.isHasRole()) {
+            throw new RuntimeException("This is not a Juniper study!");
+        }
+        HashMap<String, KitType> kitTypes = KitType.getKitLookup();
+        String key = kitTypeName + "_" + ddpInstanceDto.getDdpInstanceId();
+        KitType kitType = kitTypes.get(key);
+        if (kitType == null) {
+            return new KitResponse(UNKNOWN_KIT_TYPE, juniperKitRequest.getJuniperKitId());
+        }
+
+        Map<Integer, KitRequestSettings> kitRequestSettingsMap =
+                KitRequestSettings.getKitRequestSettings(String.valueOf(ddpInstanceDto.getDdpInstanceId()));
+        KitRequestSettings kitRequestSettings = kitRequestSettingsMap.get(kitType.getKitTypeId());
+
+        // if the kit type has sub kits > like for testBoston
```

Comment:
Unclear what is going on here.

Reply:
This is the check for subkits (multiple tubes and barcode in one package) . This is not a need right now so I commented the code so we don't spend time debugging it but kept it to for the time that we need it